DoSomething/rogue
# Filters posts powered by the API

#### What's this PR do?
- Changes Campaign Single Index page's initial state to be powered by API instead of sending signups from backend (`GET /posts`). 
- Powers filtering by status via the API (`GET /posts`).
- Adds relation to Post model of other related posts under the same signup (siblings).
- Updates Post Transformer to optionally include related posts (siblings) and signup object for `GET /posts`).
- Returns more information about the user in the User Transformer (user's last name, birthdate, email address, mobile number).  
- Returns the original photo url in the Post Transformer. 
- Updates `$include` param to allow for multiple values (e.g. `$include=signup,siblings`).
- Updates Post tests.  
- BONUS: 
    - Adds some missing things to README file for documentation.

#### How should this be reviewed?
- Test `campaigns/:nid` to make sure initial state looks as expected and filtering works properly. 
    - Make sure that all information about the signup and post is exposed correctly! 
    - Don't forget to check the "Original Photo" and "Edited Photo" links and that they're working correctly! 
- Make sure `/campaigns/:nid/inbox` isn't broken. 

#### Any background context you want to provide?
- I think that this breaks pagination? Is this ok for now and we can make another issue to fix this? 
- @DFurnes I don't think we need to do this but should we update the activity test to make sure it includes additional information about the user? Has same functionality so not sure if this is needed! 

#### Relevant tickets
Fixes https://www.pivotaltracker.com/n/projects/2019429/stories/149809538

#### Checklist
